### ISSUE TITLE
Expected hostname at index

### ISSUE THREAD

**Issue by hussainsultan** [jcrist/skein#10]
On master, when I try to submit hello world example from the documentation, I get the following error.
```
(py35) hussainsultan@cluster-07d2-m:~$ skein application submit helloworld-skein.yaml
Error: Failed to submit application, exception:
Expected hostname at index 7: file://:0/home/hussainsultan/payload.txt
```

**Comment by jcrist** [comment]
Hmmm, that's odd. Can you provide the contents of the yaml, the file tree you're running in, and what version of python you're using?

**Comment by hussainsultan** [comment]
```
hussainsultan@cluster-07d2-m:~$ skein --version
skein 0.0.2+17.gd3be11a
```
(py35) hussainsultan@cluster-07d2-m:~$ python --version
Python 3.5.5 :: Anaconda, Inc.
```
File:
```(py35) hussainsultan@cluster-07d2-m:~$ find `pwd` -name helloworld-skein.yaml 
/home/hussainsultan/helloworld-skein.yaml
```
File contents:
```
>>cat helloworld-skein.yaml 
name: hello_world
queue: default

services:
  my_service:
    resources:
      vcores: 1
      memory: 128
    files:
      - file: payload.txt
    commands:
      - echo "Sleeping for 60 seconds"
      - sleep 60
      - cat payload.txt
      - echo "Stopping service"
```

**Comment by hussainsultan** [comment]
I can confirm that without the `files:` block in the yaml the application submits just fine.

**Comment by jcrist** [comment]
Hmmm, sounds like it's a parsing issue. I can't reproduce locally. What does this produce?

```python
import skein
job = skein.Job.from_file('hello_world.yaml')
print(job.services['my_service'].files['payload.txt'].source)
```

**Comment by hussainsultan** [comment]
```In [3]: import skein
   ...: job = skein.Job.from_file('hello_world.yaml')
   ...: print(job.services['my_service'].files['/home/hussainsultan/payload.txt'].source)
   ...: 
   ...: 
   ...: 
   ...: 
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-3-387103caddff> in <module>()
      1 import skein
      2 job = skein.Job.from_file('hello_world.yaml')
----> 3 print(job.services['my_service'].files['/home/hussainsultan/payload.txt'].source)
      4 
      5 

KeyError: '/home/hussainsultan/payload.txt'

```

**Comment by jcrist** [comment]
That's not what I asked you to run (please just copy-paste the example code). The files source can be a relative or absolute path, but the destination (which is what the keys in `files` attribute are) is always relative. How about:

```python
import skein
job = skein.Job.from_file('hello_world.yaml')
for (k, v) in job.services['my_service'].files.items():
    print(k)
    print(v.source)
```

**Comment by hussainsultan** [comment]
Ahh- sorry about that. I didn't realize what `.files` was doing. Thank you for clarification.

```
In [2]: import skein
   ...: job = skein.Job.from_file('hello_world.yaml')
   ...: for (k, v) in job.services['my_service'].files.items():
   ...:     print(k)
   ...:     print(v.source)
   ...:     
payload.txt
file:///home/hussainsultan/payload.txt
```

**Comment by jcrist** [comment]
Ok. I suspect this is a hadoop version issue. Can you run the following:

```
skein daemon stop
skein daemon start --log log.txt
skein application submit hello_world.yaml
# wait for failure
skein daemon stop
```

And then post any output to stdout here, as well as the contents of `log.txt`?

**Comment by hussainsultan** [comment]
See below:

----------------------------
```bash
(ds3) hussainsultan@cluster-07d2-m:~$ skein daemon stop
(ds3) hussainsultan@cluster-07d2-m:~$ skein daemon start --log log_skein_10.txt
localhost:53524
(ds3) hussainsultan@cluster-07d2-m:~$ skein application submit hello_world.yaml
Error: Failed to submit application, exception:
Expected hostname at index 7: file://:0/home/hussainsultan/payload.txt
(ds3) hussainsultan@cluster-07d2-m:~$ skein daemon stop
```
and output from the logs:
```
(ds3) hussainsultan@cluster-07d2-m:~$ cat log_skein_10.txt 
18/05/31 01:07:30 INFO gcs.GoogleHadoopFileSystemBase: GHFS version: 1.6.6-hadoop2
18/05/31 01:07:30 INFO client.RMProxy: Connecting to ResourceManager at cluster-07d2-m/10.142.0.4:8032
18/05/31 01:07:31 INFO skein.Daemon: Server started, listening on 53524
18/05/31 01:07:37 INFO skein.Daemon: Uploading file:/home/hussainsultan/.skein/skein.crt to hdfs://cluster-07d2-m/user/hussainsultan/.skein/application_1527693992455_0037/skein.crt
18/05/31 01:07:38 INFO skein.Daemon: Uploading file:/home/hussainsultan/.skein/skein.pem to hdfs://cluster-07d2-m/user/hussainsultan/.skein/application_1527693992455_0037/skein.pem
18/05/31 01:07:38 INFO skein.Daemon: SERVICE: my_service - writing script to hdfs://cluster-07d2-m/user/hussainsultan/.skein/application_1527693992455_0037/my_service.sh
18/05/31 01:07:46 INFO skein.Daemon: Shutting down gRPC server
18/05/31 01:07:46 INFO skein.Daemon: gRPC server shut down
```
** Hadoop Version **
```bash
(ds3) hussainsultan@cluster-07d2-m:~$ hadoop version
Hadoop 2.8.3
Subversion https://bigdataoss-internal.googlesource.com/third_party/apache/bigtop -r eed50ac1b60b23428a7ce7596b96ad01d5c3a6f4
Compiled by bigtop on 2018-04-10T17:21Z
Compiled with protoc 2.5.0
From source with checksum f31cfadc2fb794479789fdcce281e07c
This command was run using /usr/lib/hadoop/hadoop-common-2.8.3.jar
```

**Comment by jcrist** [comment]
Yes, this is a version incompatibility in one of the conversion functions. I believe this to be fixed in f2bc1dfe035ecc, can you try out master?

**Comment by hussainsultan** [comment]
Fixed in master. Thank you!
